src/FastRouteRouter.php

```diff
@@ -114,12 +159,14 @@ public function match(Request $request)
         // Inject any pending routes
         $this->injectRoutes();
 
+        $dispatchData = $this->getDispatchData();
+
         $path       = $request->getUri()->getPath();
         $method     = $request->getMethod();
-        $dispatcher = $this->getDispatcher($this->router->getData());
+        $dispatcher = $this->getDispatcher($dispatchData);
         $result     = $dispatcher->dispatch($method, $path);
 
-        if ($result[0] != Dispatcher::FOUND) {
+        if ($result[0] !== Dispatcher::FOUND) {
```

Comment:
Alright. I don't like that the interface doesn't enforce them, but I suppose there's no way to do that without some massive overhead in there...